bin-src/tasks/upload.ts

```diff
@@ -153,10 +155,28 @@ export const traceChangedFiles = async (ctx: Context, task: Task) => {
   transitionTo(tracing)(ctx, task);
 
   const statsPath = join(ctx.sourceDir, ctx.fileInfo.statsPath);
-  const { changedFiles } = ctx.git;
+  const { baselineCommits, changedFiles, packageManifestChanges } = ctx.git;
   try {
+    const changedDependencyNames = await findChangedDependencies(baselineCommits).catch((e) => {
+      ctx.log.debug(e);
+    });
```

Comment:
Would it make sense to also figure out the `packageLockChanges` ahead of time and only diff those commits? [Similarly we'd need to use replacement commits when the commit was missing]

Reply:
Are you suggesting we run `findChangedDependencies` earlier on in the process, or are you suggesting to determine what baselines have lockfile changes, and feed that into `findChangedDependencies` so we don't have to check _all_ baselines?

Reply:
The latter. I only suggested it because we do that optimisation for package file changes.

Reply:
Going to punt on that for now. This PR has been dragging on for too long as it is.